src/govuk/components/input/index.js

```diff
@@ -41,10 +43,12 @@ function Input(props) {
       {errorMessageComponent}
       <input
         id={id}
+        autoFocus={autofocus}
         className={`govuk-input ${className || ''} ${
           errorMessage ? ' govuk-input--error' : ''
         }`}
         name={name || id}
+        ref={customRef}
```

Comment:
I'm a little uncomfortable with this custom extension of the params for a single component. I think the better solution for this one would be to use forwardRef as proposed in #2 

See https://reactjs.org/docs/forwarding-refs.html

Reply:
@UsmanJ I will have a crack at #2 over the next few days if I can

Reply:
@andymantell I did have a go at using forwardRef but we were trying to use the ref 3 components in and for some reason the ref was always null and didn't update. Very possible I was doing something wrong so I will also look at this again.

Reply:
Did you modify the Input component to accept a forwarded ref? With the `React.forwardRef`gubbins?

Reply:
Yes I did. I'll give it a go again later and drop a snippet in here when I do it.

Reply:
I've forwarded a ref in initially just for the Input component in #41. Does this do what you need? I think it's a sane and good addition?

I'm going to work through adding refs to the other components now... It was always the plan anyway (hence why it is logged as ticket number 2).

Reply:
I believe that will allow me to use forwardRef now. Will try it and get back to you.

Reply:
See #41 - I have added refs to more components now. Hopefully this will help... If you have time to give the code a quick scan and review that would be ace.